ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -43,28 +43,23 @@ export class JobResourcesComponent implements OnInit {
     private readonly viewContainer: ViewContainerRef) {}
 
   ngOnInit() {
-    if (this.job.inputs) {
+    if (this.job.inputs && Object.keys(this.job.inputs).length > 0) {
       this.inputs = Object.keys(this.job.inputs).sort();
       this.tabIds.push('inputs');
     }
-    if (this.job.outputs) {
+    if (this.job.outputs && Object.keys(this.job.outputs).length > 0) {
       this.outputs = Object.keys(this.job.outputs).sort();
       this.tabIds.push('outputs');
     }
 
     if (this.job.extensions) {
-      if (this.job.extensions.sourceFile) {
+      if (this.job.extensions.sourceFile && this.job.extensions.sourceFile.length > 0) {
```

Comment:
This addition doesn't look necessary me; empty string is falsy in JS -  unless typescript is doing something I'm not aware of.